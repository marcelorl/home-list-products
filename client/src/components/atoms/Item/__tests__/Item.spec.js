import createTestContext from 'react-cosmos-test/enzyme'

import fixture from '../Item.fixture'

const { mount, getWrapper, get } = createTestContext({fixture})

beforeEach(mount)

describe('#Item atom', () => {
  it('renders component', () => {
    expect(getWrapper()).toMatchSnapshot()
  })

  it('has all data passed through props', () => {
    expect(get('props').brandName).toEqual(getWrapper('.card-title').text())
    expect(get('props').description).toEqual(getWrapper('.card-subtitle').text())
    expect(get('props').priceFormatted).toContain(getWrapper('.item__price').text())
    expect('12x 83.33 â‚¬').toContain(getWrapper('.item__parcels').text())
  })
})
