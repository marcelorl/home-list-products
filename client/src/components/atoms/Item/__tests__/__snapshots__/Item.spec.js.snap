// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Item atom renders component 1`] = `
<Item
  brandName="brandName"
  description="description"
  image="test.png"
  price={1000}
  priceFormatted="1000 R$"
>
  <styled.div>
    <div
      className="sc-bdVaJa hNlXVo"
    >
      <Card
        tag="div"
      >
        <div
          className="card"
        >
          <Styled(CardImg)
            src="test.png"
            top={true}
            width="100%"
          >
            <CardImg
              className="sc-bwzfXH cDhPkI"
              src="test.png"
              tag="img"
              top={true}
              width="100%"
            >
              <img
                className="sc-bwzfXH cDhPkI card-img-top"
                src="test.png"
                width="100%"
              />
            </CardImg>
          </Styled(CardImg)>
          <CardBody
            tag="div"
          >
            <div
              className="card-body"
            >
              <Styled(CardTitle)>
                <CardTitle
                  className="sc-htpNat kAvNiS"
                  tag="h5"
                >
                  <h5
                    className="sc-htpNat kAvNiS card-title"
                  >
                    brandName
                  </h5>
                </CardTitle>
              </Styled(CardTitle)>
              <Styled(CardSubtitle)>
                <CardSubtitle
                  className="sc-bxivhb dyXxrV"
                  tag="h6"
                >
                  <h6
                    className="sc-bxivhb dyXxrV card-subtitle"
                  >
                    description
                  </h6>
                </CardSubtitle>
              </Styled(CardSubtitle)>
              <div>
                <div
                  className="item__price"
                >
                  1000 R$
                </div>
                <div
                  className="item__parcels"
                >
                  12x 83.33 â‚¬
                </div>
              </div>
            </div>
          </CardBody>
        </div>
      </Card>
    </div>
  </styled.div>
</Item>
`;
